src/assets/wise5/authoringTool/project-authoring/project-authoring.component.ts

```diff
@@ -147,16 +148,15 @@ export class ProjectAuthoringComponent {
    * The checkbox for a node was clicked. We do not allow selecting a mix of group and step nodes.
    * If any group nodes are selected, disable all step node checkboxes, and vise-versa.
    */
-  protected selectNode(): void {
-    const checkedNodes = this.items
-      .concat(Object.values(this.idToNode))
-      .filter((item) => item.checked);
+  protected selectNode({ id, checked }: SelectNodeEvent): void {
+    this.nodeIdToChecked[id] = checked;
+    const checkedNodes = Object.entries(this.nodeIdToChecked).filter((entry) => entry[1]);
     if (checkedNodes.length === 0) {
-      this.groupNodeSelected = false;
-      this.stepNodeSelected = false;
+      this.setNodeTypeSelected(null);
     } else {
-      this.groupNodeSelected = this.isGroupNode(checkedNodes[0].id);
-      this.stepNodeSelected = !this.groupNodeSelected;
+      this.setNodeTypeSelected(
+        this.isGroupNode(checkedNodes[0][0]) ? NodeTypeSelected.lesson : NodeTypeSelected.step
```

Comment:
Change isGroupNode() to take in the node instead of id? Also, change isGroupNode()'s access modifier to private instead of protected now that it's only being called from this class?
```suggestion
        this.isGroupNode(checkedNodes[0]) ? NodeTypeSelected.lesson : NodeTypeSelected.step
```

Reply:
You only have the node id here, not the node.